ffi/derive/src/convert.rs

```diff
@@ -490,3 +529,30 @@ fn is_opaque_wrapper(input: &DeriveInput) -> bool {
     let opaque_attr = parse_quote! {#[opaque_wrapper]};
     input.attrs.iter().any(|a| *a == opaque_attr)
 }
+
+fn add_bounds_and_split_generics<'generics>(
+    generics: &'generics mut syn::Generics,
+    lifetimes: &'generics [syn::Lifetime],
+    trait_bounds: &'generics [syn::TraitBound],
+) -> (
+    syn::punctuated::Punctuated<syn::GenericParam, syn::Token![,]>,
+    syn::TypeGenerics<'generics>,
+    Option<&'generics syn::WhereClause>,
+) {
+    generics
+        .lifetimes_mut()
+        .for_each(|lt| lt.bounds.extend(lifetimes.iter().cloned()));
+    generics.type_params_mut().for_each(|param| {
+        param
+            .bounds
+            .extend(lifetimes.iter().cloned().map(syn::TypeParamBound::from));
```

Comment:
```suggestion
        param.bounds.extend(parse_quote! {#lifetimes});
```

if you could do it this way, I think it'll be more readable